src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts

```diff
@@ -250,12 +255,28 @@ export const thorchainSaversStakingOpportunitiesUserDataResolver = async ({
       const rewardsAmountsCryptoBaseUnit: [string] = [
         stakedAmountCryptoBaseUnitIncludeRewards.minus(stakedAmountCryptoBaseUnit).toFixed(),
       ]
+      let maturity: number | undefined = undefined
+
+      if (stakingOpportunityId === thorchainAssetId && accountPosition.last_add_height) {
+        const blockParams = new URLSearchParams({
+          height: accountPosition.last_add_height?.toString(),
+        })
+
+        const { data: blockDetails } = await axios.get<ThorchainBlock>(
+          `${getConfig().REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/block?${blockParams}`,
+        )
+
+        maturity =
+          new Date(blockDetails.header.time).getTime() +
+          thorchainBlockTimeMs * RUNEPOOL_MINIMUM_WITHDRAW_BLOCKS
+      }
 
       stakingOpportunitiesUserDataByUserStakingId[userStakingId] = {
         isLoaded: true,
         userStakingId,
         stakedAmountCryptoBaseUnit: stakedAmountCryptoBaseUnit.toFixed(),
         rewardsCryptoBaseUnit: { amounts: rewardsAmountsCryptoBaseUnit, claimable: false },
+        maturity,
```

Comment:
Re: optional vs. undefined, given `maturity` is the current discriminator in `RunepoolUserStakingOpportunity`, do we want to optionally pass this vs. passing `maturity | undefined`? i.e something like this

```suggestion
	...(maturity ? { maturity } : {}),
```

Reply:
Fixed in [231aabd](https://github.com/shapeshift/web/pull/7472/commits/231aabd2822d7bf58613c7fd32ba5873c67846a8)